docs/self-managed/setup/deploy/azure/azure-aks/terraform-setup.md

```diff
@@ -0,0 +1,465 @@
+
+##### terraform_sp_app_id
+
+This is the Application (client) ID of the Azure Service Principal that Terraform uses for authentication. It is required to configure access policies in services such as Key Vault, particularly when using customer-managed keys (CMK) with AKS.
+
+If you created a Service Principal manually, you can retrieve its application ID with the following command:
+
+```shell
+az ad sp list --display-name "<your-service-principal-name>" --query "[0].appId" -o tsv
+```
+
+If you're already using a Service Principal to authenticate (for example, with `az login --service-principal`), this value corresponds to the `appId` you supplied during login.
+
+This value is critical because Terraform uses it to assign the necessary permissions for interacting with encryption keys and other protected resources. If the ID is incorrect or omitted, key-related configurations may fail, and AKS will be unable to use CMK for securing cluster secrets.
+
+#### Initialize Terraform
+
+Once your authentication is set up, you can initialize your Terraform project. The previous steps configured a dedicated Azure Storage Account and container (`AZURE_STORAGE_ACCOUNT_NAME`, `AZURE_STORAGE_CONTAINER_NAME`) to store your state, and the following creates a backend key that will be used by your configuration.
+
+Configure the backend and download the necessary provider plugins:
+
+```bash reference
+https://github.com/camunda/camunda-deployment-references/blob/main/azure/common/procedure/storage-account/storage-account-tf-init.sh
+```
+
+Terraform will connect to the Azure storage container to manage the state file, ensuring remote and persistent storage.
+
+### AKS cluster module setup
+
+This module establishes the foundational configuration for Azure access and Terraform.
+
+We will utilize [Terraform modules](https://developer.hashicorp.com/terraform/language/modules), which allow us to abstract resources into reusable components, streamlining our infrastructure management and following Terraform best practices.
+
+The reference architecture comes with an example module implementation of the [AKS cluster](https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks-cluster/) and offers a robust starting point for deploying an AKS cluster. It is highly recommended to review this module prior to implementation to understand its structure and capabilities.
+
+The module will be locally sourced, meaning within your cloned repository you can make adjustments to the AKS module directly, and the changes will immediately apply to your setup.
+
+#### Set up the AKS cluster module
+
+1. The `cluster.tf` in your chosen reference contains a basic setup referencing a local Terraform module with the cluster configuration. The following shows said file, which you can adjust within your cloned setup to suit your needs.
+
+```hcl reference
+https://github.com/camunda/camunda-deployment-references/blob/main/azure/modules/aks/cluster.tf
+```
```

Comment:
please double check here what you actually do and not just copy and paste everything.

Reply:
current state makes no sense

Reply:
it would make sense to divdie it up into `cluster.tf` and other use cases.

Reply:
the idea in EKS is that you have the reference that uses the modules.
Yes you can eventually adjust the modules if you need something specific but not adjust it for the base setup to configure it.
You wouldn't hardcode the value in a module to adjust it. Imagine you were using the same for multiple clusters as base and need differences for each setup.

The context in EKS is a completely different one, we're only ever targeting the reference itself never adjust the module under it.
Which is the suggestion you're doing here. Also the `cluster.tf` as said does not exist in the reference but only in the module.